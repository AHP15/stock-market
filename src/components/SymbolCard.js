/* eslint-disable object-curly-newline */
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import styles from '../styles/SymbolCard.module.css';

import more from '../assets/arrow.png';

// eslint-disable-next-line arrow-body-style
const SymbolCard = ({ symbol, company, price, dark, index }) => {
  return (
    <Link
      to={`/details/${symbol}/${index + 1}`}
      className={styles[`${dark ? 'dark' : 'link'}`]}
    >
      <h3 className={styles.symbol}>{symbol}</h3>
      <img src={more} alt="details" className={styles.more} />
      <div className={styles.info}>
        <h4>{company}</h4>
        <p>
          {price}
          $
        </p>
      </div>
    </Link>
  );
};

SymbolCard.propTypes = {
  symbol: PropTypes.string.isRequired,
  company: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  dark: PropTypes.bool.isRequired,
  index: PropTypes.number.isRequired,
};

export default SymbolCard;
