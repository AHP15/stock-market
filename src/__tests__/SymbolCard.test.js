import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';

import SymbolCard from '../components/SymbolCard';

import Store from '../slices/store';

it('renders Details correctly', () => {
  const tree = renderer.create(
    <BrowserRouter>
      <Provider store={Store}>
        <SymbolCard symbol="test" company="test" price={100} dark={false} index={0} />
      </Provider>
    </BrowserRouter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
