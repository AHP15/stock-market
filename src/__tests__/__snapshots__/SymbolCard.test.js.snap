// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Details correctly 1`] = `
<a
  className="link"
  href="/details/test/1"
  onClick={[Function]}
>
  <h3
    className="symbol"
  >
    test
  </h3>
  <img
    alt="details"
    className="more"
    src="arrow.png"
  />
  <div
    className="info"
  >
    <h4>
      test
    </h4>
    <p>
      100
      $
    </p>
  </div>
</a>
`;
